### ISSUE TITLE
Problem with Mixed Content (HTTP/HTTPS) in TheoryAndPractice

### ISSUE THREAD

**Issue by pablodecm** [cranmer/TheoryAndPractice#18]
Hi Kyle @cranmer, I was actually in the process of updating my old Pelican webpage and remember that you also had a Pelican-based webpage. So I checked our your page and repository to see how you have dealt with the Python environment updates and what were you using now.

While checking it out, I discovered that there is a problem with mixed content in that basically prevents people with the default configuration in modern browsers to access  https://theoryandpractice.org as a properly styled page, see example screenshot:

![image](https://github.com/cranmer/TheoryAndPractice/assets/6707437/e075088f-c5e7-488f-8c7a-b4501925ad9b)

I think it should be relatively easy to fix, mainly the I believe only required change is changed the external dependencies from the same root to https:// instead of http://.

![image](https://github.com/cranmer/TheoryAndPractice/assets/6707437/7f5620d9-652c-428c-8e90-2ae470d13800)

Not sure wether this was caused by a environment configuration in the last deployment or is actually some problem due to the legacy Python libraries or environment, but I just wanted to tell you in case you were not aware.

**Comment by cranmer** [comment]
Thanks for letting me know.

I need to find some time to update this page in general. Considering moving to a different solution, but that’s also a pain.

Kyle

From: Pablo de Castro ***@***.***>
Date: Tuesday, January 30, 2024 at 4:35 AM
To: cranmer/TheoryAndPractice ***@***.***>
Cc: Kyle Stuart Cranmer ***@***.***>, Mention ***@***.***>
Subject: [cranmer/TheoryAndPractice] Problem with Mixed Content (HTTP/HTTPS) in TheoryAndPractice (Issue #18)

Hi Kyle @cranmer<https://github.com/cranmer>, I was actually in the process of updating my old Pelican webpage and remember that you also had a Pelican-based webpage so I checked our your page and repository to see how you have dealt with the Python environment updates and what were you using now.

While checking it out, I discovered that there is a problem with mixed content in that basically prevents people with the default configuration in modern browsers to access https://theoryandpractice.org as properly styled page, see example:

image.png (view on web)<https://github.com/cranmer/TheoryAndPractice/assets/6707437/e075088f-c5e7-488f-8c7a-b4501925ad9b>

I think it should be relatively easy to fix, mainly the I believe only required change is changed the external dependencies from the same root to https:// instead of http://.

image.png (view on web)<https://github.com/cranmer/TheoryAndPractice/assets/6707437/7f5620d9-652c-428c-8e90-2ae470d13800>

Not sure wether this was caused by a environment configuration in the last deployment or is actually some problem due to the legacy Python libraries or environment, but I just wanted to tell you in case you were not aware.

**Comment by pablodecm** [comment]
Indeed, moving to a different solution can be some work if you are using different plugins and you have a significant amount of content (and you do).

I have not tried it yet, but I have heard good things about https://quarto.org/ and they support all (or at least) of the things you would want out the box I think.

Probably you thought about this, but a poor man's solution (or rather a busy man's solution) that I estimate that will take you less than 15 mins and get the page https://theoryandpractice.org back to working order would be:

- cloning https://github.com/cranmer/cranmer.github.io
- do a search and replace in all html files http:// by https://  this can be done with VSCode for example.
- pushing it and it gets deployed

Have a good day!
